perllib/FixMyStreet/App/Controller/Admin/Roles.pm

```diff
@@ -55,7 +55,22 @@ sub item :PathPart('admin/roles') :Chained :CaptureArgs(1) {
 
 sub edit :PathPart('') :Chained('item') :Args(0) {
     my ($self, $c) = @_;
-    return $self->form($c, $c->stash->{obj});
+
+    my $role = $c->stash->{obj};
+
+    $c->stash->{body} = $role->body;
+    $c->forward('/admin/fetch_contacts');
+    $c->forward('/admin/stash_contacts_for_template', [
+        \@{$role->get_extra_metadata('categories') || []},
+    ]);
+    $c->forward('/report/stash_category_groups', [ $c->stash->{contacts}, { combine_multiple => 1 } ]);
+
+    my @live_contacts = $c->stash->{live_contacts}->all;
+    my @live_contact_ids = map { $_->id } @live_contacts;
+    my @new_contact_ids = grep { $c->get_param("contacts[$_]") } @live_contact_ids;
+    $c->stash->{new_contact_ids} = \@new_contact_ids;
```

Comment:
I don't totally get why this needs to perform this check given that the values in contacts[] comes for live_contacts in the first place, so maybe a comment to explain what's happening here would be nice.

Reply:
Yeah having this code here was a bit confusing. I've moved it to where it's actually being used in 37106224520e77ed3f1d3e2abed7454f4be3b8c8 and added a comment, so hopefully that makes things a bit clearer.